apps/music/js/endpoint.js

```diff
@@ -300,51 +289,27 @@ function getArtists() {
   });
 }
 
-function getArtist(filePath) {
-  return getSong(filePath).then((song) => {
-    var artist = song.metadata.artist;
-
-    return new Promise((resolve) => {
-      Database.enumerateAll('metadata.artist', artist, 'next', (songs) => {
-        songs.sort((a, b) => {
-          var albumSort = a.metadata.album < b.metadata.album ?
-            -1 : (a.metadata.album > b.metadata.album ?  1 : 0);
-
-          return albumSort || (a.metadata.discnum - b.metadata.discnum) ||
-                              (a.metadata.tracknum - b.metadata.tracknum);
-        });
-
-        resolve(songs);
-      });
-    });
-  });
+function getAlbums() {
+  return Database.albums();
 }
 
-function getPlaylists() {
-  return Promise.resolve(Database.playlists);
+function getSongs() {
+  return Database.songs();
 }
 
-function getPlaylist(id) {
-  var playlist = Database.playlists.find(playlist => playlist.id === id);
-
-  return new Promise((resolve) => {
-    Database.enumerateAll(playlist.index, null, playlist.direction, (songs) => {
-      resolve(songs);
-    });
-  });
+function getSongCount() {
+  return Database.totalCount();
 }
 
-function getSongs() {
-  return new Promise((resolve) => {
-    Database.enumerateAll('metadata.title', null, 'next', (songs) => {
-      resolve(songs);
-    });
+function getArtist(filePath) {
+  return getSong(filePath).then((track) => {
```

Comment:
`song` is probably the right thing for now, but my own OCD makes me annoyed when I see us use "song", since a single file isn't necessarily a song (it could be a podcast, or a full concert, etc).